PrincessPaperplane/discordBot.py

```diff
@@ -304,89 +316,97 @@ async def on_message(message):
                 cur.execute("INSERT INTO user_info (`id`, `name`, `exp`, `expTime`, `avatar_url`) VALUES (%s, %s, %s, %s, %s)", (author.id, author.name, exp, time.time(), author.avatar_url, ))
             if server.id == TEST_SERVER:
                 cur.execute("INSERT INTO user_info_test (`id`, `name`, `exp`, `expTime`, `avatar_url`) VALUES (%s, %s, %s, %s, %s)", (author.id, author.name, exp, time.time(), author.avatar_url,))
+    
+    # Handle commands
+    match = DICE_CMD_REGEX.match(message.content)
+    if bool(match):
+        cmd_length = len(match.group(1)) + len(match.group(2))
+        message.content = message.content[:cmd_length] + " " + message.content[cmd_length:]
+    await bot.process_commands(message)
+
+#Used as decorator
+def is_in_channel(channel_id : int):
+    """Used as decorator that checks if bot can post in this channel
+    """
+    #Define predicate to be checked
+    async def predicate(ctx):
+        if ctx.server.id == LIVE_SERVER and ctx.channel.id != channel_id: # only allow !rank / !rang in #bod_spam
+            return False
+        else:
+            return True
+    # Return as check, use with decorator
+    return commands.check(predicate)
+
+@commands.command(aliases=['rank', 'rang'])

```

Comment:
Wirft den Fehler `discord.ext.commands.errors.CommandNotFound: Command "rank" is not found`
Stattdessen hat `@bot.command` funktioniert